src/shell/shell.cpp

```diff
@@ -736,6 +754,99 @@ class AUTOEXEC final : public Module_base {
 	}
 };
 
+// This function takes in a conf file having preferences representing
+// arguments to the "mount" command, but in typical conf format:
+//    [drive]
+//    type = dir / cdrom / floppy
+//    label = string
+//
+// As well as a couple more settings:
+// path = R:\DIR;%PATH%;..., which can be used to add directories to the
+//    PATH in any manner or order the user wishes.
+//
+// verbose = true / false ... which either shows or hides the output from
+//    the mount command.
+//
+// The function returns a pair: "mount" command arguments and the
+// path string, both of which might be empty.
+//
+static std::pair<std::string, std::string> parse_drive_conf(const std_fs::path &conf_path)
+{
+	// If the conf path doesn't exist, at least return the default quiet arg
+	constexpr auto quiet_arg      = " > NUL";
+	constexpr auto empty_path_val = "";
+	if (!path_exists(conf_path))
+		return {quiet_arg, empty_path_val};
+
+	// A conf file exists, so we prepare our config parser.
+	auto cmd  = CommandLine(0, "");
+	auto conf = Config(&cmd);
+
+	constexpr auto changeable_at_runtime = false;
+	auto prop = conf.AddSection_prop("drive", nullptr, changeable_at_runtime);
```

Comment:
I'd suggest extracting the parsing part out into a separate function that just takes the config as a string. This would also aid testability; currently this function is doing a bit too much.

Reply:
It's now split into three portions:

1. Creating a `Config` object and defining the section, fields, and values we expect in the config. 

2. Parsing the file by the Config object (and let default broken values)

3. Making sense of the values and transforming them into mount-line parameters.